File: backend-python/app/schemas/spare_parts.py
from datetime import datetime

from pydantic import BaseModel, Field


class SparePartsUsageCreate(BaseModel):
    product_name: str = Field(..., min_length=1, max_length=200, description="产品名称")
    brand: str | None = Field(None, max_length=100, description="品牌")
    model: str | None = Field(None, max_length=100, description="产品型号")
    quantity: int = Field(..., gt=0, description="领用数量")
    user_name: str = Field(..., min_length=1, max_length=100, description="运维人员")
    issue_time: str | datetime = Field(..., description="领用时间")
    unit: str = Field("件", max_length=20, description="单位")
    project_id: str | None = Field(None, max_length=50, description="项目编号")
    project_name: str | None = Field(None, max_length=200, description="项目名称")
    remark: str | None = Field(None, max_length=500, description="备注")
    stock_id: int | None = Field(None, description="库存记录ID")
    status: str = Field("待归还", max_length=20, description="状态")


class SparePartsUsageUpdate(BaseModel):
    product_name: str = Field(..., max_length=200, description="产品名称")
    brand: str | None = Field(None, max_length=100, description="品牌")
    model: str | None = Field(None, max_length=100, description="产品型号")
    quantity: int = Field(..., description="领用数量")
    user_name: str = Field(..., max_length=100, description="运维人员")
    issue_time: datetime = Field(..., description="领用时间")
    unit: str = Field("件", max_length=20, description="单位")
    project_id: str | None = Field(None, max_length=50, description="项目编号")
    project_name: str | None = Field(None, max_length=200, description="项目名称")
    stock_id: int | None = Field(None, description="库存记录ID")
    status: str = Field("待归还", max_length=20, description="状态")

File: H5/src/types/models/spareParts.ts
/**
 * 备件数据模型
 */

/**
 * 备件库存
 */
export interface SparePartsStock {
  id: number
  product_name: string
  productName?: string
  brand?: string
  model?: string
  unit: string
  quantity: number
  stock?: number
  status: string
  created_at: string
  updated_at: string
}

/**
 * 备件使用记录
 */
export interface SparePartsUsage {
  id: number
  product_name: string
  productName?: string
  brand?: string
  model?: string
  quantity: number
  return_quantity?: number
  user_name: string
  userName?: string
  issue_time: string
  issueTime?: string
  unit: string
  project_id?: string
  project_name?: string
  projectName?: string
  stock_id?: number | null
  status: string
  created_at: string
  updated_at: string
}

/**
 * 备件入库记录
 */
export interface SparePartsInbound {
  id: number
  inbound_no: string
  inboundNo?: string
  product_name: string
  productName?: string
  brand?: string
  model?: string
  quantity: number
  supplier?: string
  unit: string
  user_name: string
  userName?: string
  remarks?: string
  inbound_time?: string
  inboundTime?: string
  created_at: string
}

/**
 * 备件领用请求
 */
export interface SparePartsIssueRequest {
  stock_id?: number
  product_name?: string
  brand?: string | null
  model?: string | null
  quantity: number
  user_name?: string
  issue_time?: string
  unit?: string
  project_id?: string | null
  project_name?: string | null
}

/**
 * 备件归还请求
 */
export interface SparePartsReturnRequest {
  usage_id: number
  quantity: number
}

/**
 * 备件入库请求
 */
export interface SparePartsInboundRequest {
  product_name: string
  brand?: string
  model?: string
  quantity: number
  supplier?: string
  unit: string
  user_name?: string
  remarks?: string
}

/**
 * 备件库存查询参数
 */
export interface SparePartsStockQueryParams {
  page?: number
  size?: number
  pageSize?: number
  product_name?: string
  brand?: string
}

/**
 * 备件使用查询参数
 */
export interface SparePartsUsageQueryParams {
  page?: number
  size?: number
  pageSize?: number
  product_name?: string
  user_name?: string
  project_name?: string
  status?: string
}
